include/mbgl/map/backend.hpp

```diff
@@ -20,6 +20,9 @@ class Backend {
     // Returns the backend's context which manages OpenGL state.
     gl::Context& getContext();
 
+    using glProc = void (*)();
+    void initializeExtensions(glProc (*getProcAddress)(const char*));
```

Comment:
So far, that "ideal time" is when the backend implementors call `initializeExtensions`. We'd still have to retain that call, but we could remove the `getProcAddress` parameter. Not sure that it's worth it, tbh